common/src/chain/transaction/signature/tests/mod.rs

```diff
@@ -0,0 +1,339 @@
+        let signature = match tx_updater.inputs[0].get_witness() {
+            InputWitness::Standard(signature) => {
+                // Let's change around 20ish last bytes, it's also avoided SCALE errors
+                let mut raw_signature = (&signature.get_raw_signature()[0..60]).to_vec();
+                let body_signature: Vec<u8> = signature
+                    .get_raw_signature()
+                    .iter()
+                    .skip(60)
+                    .map(|item| {
+                        if item < &u8::MAX {
+                            item.wrapping_add(1)
+                        } else {
+                            item.wrapping_sub(1)
+                        }
+                    })
+                    .collect();
+
+                raw_signature.extend(body_signature);
+                StandardInputSignature::new(signature.sighash_type(), raw_signature)
+            }
+            InputWitness::NoSignature(_) => unreachable!(),
+        };
+        tx_updater.inputs[input_num].update_witness(InputWitness::Standard(signature));
+        let tx = tx_updater.generate_tx().unwrap();
+        assert_eq!(
+            verify_signature(outpoint_dest, &tx, input_num),
+            Err(TransactionSigError::SignatureVerificationFailed)
+        );
+    }
+}
+
+fn check_insert_output(original_tx: &Transaction, destination: &Destination, should_fail: bool) {
+    let mut tx_updater = MutableTransaction::from(original_tx);
+    let (_, pub_key) = PrivateKey::new(KeyKind::RistrettoSchnorr);
+    tx_updater.outputs.push(TxOutput::new(
+        Amount::from_atoms(1234567890),
+        Destination::PublicKey(pub_key),
+    ));
+    let tx = tx_updater.generate_tx().unwrap();
+    assert_verify_signature(destination, &tx, should_fail);
+}
+
+fn check_change_output(original_tx: &Transaction, destination: &Destination, should_fail: bool) {
+    // Should failed due to change in output value
+    let mut tx_updater = MutableTransaction::from(original_tx);
+    tx_updater.outputs[0] = TxOutput::new(
+        (tx_updater.outputs[0].get_value() + Amount::from_atoms(100)).unwrap(),
+        tx_updater.outputs[0].get_destination().clone(),
+    );
+    let tx = tx_updater.generate_tx().unwrap();
+    assert_verify_signature(destination, &tx, should_fail);
+}
+
+fn check_change_input(original_tx: &Transaction, destination: &Destination, should_fail: bool) {
+    // Should failed due to change in output value
+    let mut tx_updater = MutableTransaction::from(original_tx);
+    tx_updater.inputs[0] = TxInput::new(
+        OutPointSourceId::Transaction(Id::<Transaction>::from(H256::random())),
+        9999,
+        tx_updater.inputs[0].get_witness().clone(),
+    );
+    let tx = tx_updater.generate_tx().unwrap();
+    assert_verify_signature(destination, &tx, should_fail);
+}
+
+fn assert_verify_signature(outpoint: &Destination, tx: &Transaction, should_fail: bool) {
```

Comment:
This function can be removed. It isn't necessary and just complicates reading the code. Just use an unwrap with an assert.